examples/src/main/java/ai/djl/examples/training/transferlearning/TransferFreshFruitCustomizeSampler.java

```diff
@@ -0,0 +1,289 @@
+                    Model model = trainer.getModel();
+                    float accuracy = result.getValidateEvaluation("Accuracy");
+                    model.setProperty("Accuracy", String.format("%.5f", accuracy));
+                    model.setProperty("Loss", String.format("%.5f", result.getValidateLoss()));
+                });
+
+        return new DefaultTrainingConfig(new SoftmaxCrossEntropy("SoftmaxCrossEntropy"))
+                .addEvaluator(new Accuracy())
+                .optDevices(Engine.getInstance().getDevices(1))
+                .addTrainingListeners(TrainingListener.Defaults.logging(outputDir))
+                .addTrainingListeners(listener);
+    }
+
+    private static class SoftmaxCrossEntropy extends Loss {
+
+        /**
+         * Base class for metric with abstract update methods.
+         *
+         * @param name The display name of the Loss
+         */
+        public SoftmaxCrossEntropy(String name) {
+            super(name);
+        }
+
+        /** {@inheritDoc} */
+        @Override
+        public NDArray evaluate(NDList labels, NDList predictions) {
+            // Here the labels are supposed to be one-hot
+            int classAxis = -1;
+            NDArray pred = predictions.singletonOrThrow().log();
+            NDArray lab = labels.singletonOrThrow().reshape(pred.getShape());
+            NDArray loss = pred.mul(lab).neg().sum(new int[] {classAxis}, true);
+            return loss.mean();
+        }
+    }
+
+    private static class BatchRandomSamplerWithCut implements Sampler, AutoCloseable {
+
+        private int batchSize;
+        private long cut = Long.MAX_VALUE;
+        private NDManager manager = NDManager.newBaseManager();
+
+        BatchRandomSamplerWithCut(int batchSize) {
+            this.batchSize = batchSize;
+        }
+
+        BatchRandomSamplerWithCut(int batchSize, long cut) {
+            this.batchSize = batchSize;
+            this.cut = cut;
+        }
+
+        /**
+         * Fetches an iterator that iterates through the given {@link RandomAccessDataset} in
+         * mini-batches of indices.
+         *
+         * @param dataset the {@link RandomAccessDataset} to sample from
+         * @return an iterator that iterates through the given {@link RandomAccessDataset} in
+         *     mini-batches of indices
+         */
+        @Override
+        public Iterator<List<Long>> sample(RandomAccessDataset dataset) {
+            return new Iterate(manager, dataset.size(), batchSize, cut);
+        }
+
+        /**
+         * Returns the batch size of the {@code Sampler}.
+         *
+         * @return the batch size of the {@code Sampler}, -1 if batch size is not fixed
+         */
+        @Override
+        public int getBatchSize() {
+            return batchSize;
+        }
+
+        @Override
+        public void close() {
+            manager.close();
+            manager = null;
+        }
+
+        static class Iterate implements Iterator<List<Long>> {
```

Comment:
Can you use either `datasetBuilder.optLimit(limit)` or `dataset.subDataset(0, limit)` to get the limited dataset?